CHANGELOG.md

```diff
@@ -1,5 +1,13 @@
 # CHANGELOG
 
+* Failing HTTP requests are now retried using an exponential backoff. The
+  period to keep retrying any failed request is set 5 minutes by default and
+  can be adjusted using the new `RetryFor` configuration option.
```

Comment:
Do we need to expose it in the config and the API? A sensible default would probably be fine. Other Go packages do that for instance (e.g. all the Google packages).

Reply:
The defaults do seem sensible to me.

We can remove the ability to override the defaults, but to me it did seem like something potentially useful in certain environments.